File: HomeTask/app/src/main/java/com/example/hometask/Adapter.java
package com.example.hometask;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.sip.SipAudioCall;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.io.File;

public class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {

    private String[] arraySurname;
    private String[] arrayName;
    private String[] arrayPatronymic;
    private String[] arrayPosition;
    private String[] arrayDate;
    private int[] arrayId;
    private Listener listener;

    interface Listener{
        void onClick(int i);
    }

    public Adapter(String[] arraySurname, String[] arrayName, String[] arrayPatronymic, String[] arrayPosition, String[] arrayDate, int[] arrayId) {
        this.arraySurname = arraySurname;
        this.arrayName = arrayName;
        this.arrayPatronymic = arrayPatronymic;
        this.arrayPosition = arrayPosition;
        this.arrayDate = arrayDate;
        this.arrayId = arrayId;
    }

    @Override
    public Adapter.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        CardView cardView = (CardView) LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_client, null);
        return new  ViewHolder(cardView);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, final int i) {
        final CardView cardView = viewHolder.cardView;
        ImageView image = (ImageView) cardView.findViewById(R.id.image);
        image.setImageBitmap(Image("/data/data/com.example.hometask/app_imageDir", String.valueOf(arrayId[i])));
        image.setContentDescription(arrayName[i]);
        TextView fio = (TextView) cardView.findViewById(R.id.fio);
        fio.setText(arraySurname[i] + " " + arrayName[i] + " " + arrayPatronymic[i]);
        final TextView position = (TextView) cardView.findViewById(R.id.position);
        position.setText(arrayPosition[i]);
        TextView date = (TextView) cardView.findViewById(R.id.date);
        date.setText(arrayDate[i]);
        cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null){
                    listener.onClick(arrayId[i]);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return arrayName.length;
    }

    public void setListener(Listener listener){
        this.listener = listener;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private CardView cardView;

        public ViewHolder(CardView itemView) {
            super(itemView);
            cardView = itemView;
        }
    }

    private Bitmap Image(String directory, String name){
        File file = new File(directory, name + ".png");
        Bitmap bitmap = BitmapFactory.decodeFile(file.getAbsolutePath());
        return bitmap;
    }
}

File: HomeTask/app/src/main/java/com/example/hometask/Adapter.kt
package com.example.hometask

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import java.io.File

class Adapter(private val arraySurname: Array<String?>, private val arrayName: Array<String?>, private val arrayPatronymic: Array<String?>, private val arrayPosition: Array<String?>, private val arrayDate: Array<String?>, private val arrayId: IntArray) : RecyclerView.Adapter<Adapter.ViewHolder>() {
    private var listener: Listener? = null

    interface Listener {
        fun onClick(i: Int)
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): Adapter.ViewHolder {
        val cardView = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_client, null) as CardView
        return ViewHolder(cardView)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val cardView = viewHolder.cardView
        val image = cardView.findViewById<View>(R.id.image) as ImageView
        image.setImageBitmap(Image("/data/data/com.example.hometask/app_imageDir", arrayId[i].toString()))
        image.contentDescription = arrayName[i]
        val fio = cardView.findViewById<View>(R.id.fio) as TextView
        fio.text = arraySurname[i] + " " + arrayName[i] + " " + arrayPatronymic[i]
        val position = cardView.findViewById<View>(R.id.position) as TextView
        position.text = arrayPosition[i]
        val date = cardView.findViewById<View>(R.id.date) as TextView
        date.text = arrayDate[i]
        cardView.setOnClickListener {
            if (listener != null) {
                listener!!.onClick(arrayId[i])
            }
        }
    }

    override fun getItemCount(): Int {
        return arrayName.size
    }

    fun setListener(listener: Listener) {
        this.listener = listener
    }

    class ViewHolder(val cardView: CardView) : RecyclerView.ViewHolder(cardView)

    private fun Image(directory: String, name: String): Bitmap {
        val file = File(directory, "$name.png")
        return BitmapFactory.decodeFile(file.absolutePath)
    }
}
